components/tab-list/tab-list.js

```diff
@@ -35,27 +35,55 @@ export class TabList extends HTMLElement {
 
     load() {
         return new Promise(resolve => {
-            requestAnimationFrame( async () => {
+            requestAnimationFrame(async () => {
                 this.shadowRoot.addEventListener("click", this.#clickHandler);
-                this.#target = this.#target || document.querySelector(this.getAttribute("for"));
                 await crsbinding.translations.parseElement(this);
+                // When the component is loaded, we don't have a target, so we need to find it.
 
-                await crs.call("component", "notify_ready", { element: this });
+                await this.#setInitialTab();
+
+                await crs.call("component", "notify_ready", {element: this});
                 resolve();
             })
         })
     }
 
+    async #setInitialTab() {
+        const tab = this.dataset.default;
+        const target = this.querySelector(`[data-view="${tab}"]`) || this.querySelector(`[data-view]`);
+        await this.#setSelected(tab, target);
+    }
+
     async #clicked(event) {
         const target = event.composedPath()[0];
-        await this.selectTab(target.dataset.view, target);
+        const tab = target.dataset.view;
+        await crs.call("dom_collection", "toggle_selection", {target});
+        await this.#setHiddenState(tab);
+        this.dispatchEvent(new CustomEvent("change", {detail: {tab}}));
+    }
+
+    async #setSelected(tab, target) {
+        await crs.call("dom_collection", "toggle_selection", {target});
+        await this.#setHiddenState(tab);
     }
 
-    async selectTab(tab, target) {
-        target = target || this.querySelector(`[data-view="${tab}"]`);
-        this.#target.view = tab;
-        await crs.call("dom_collection", "toggle_selection", { target });
-        this.dispatchEvent(new CustomEvent("change", { detail: { tab } }));
+    /**
+     * @method #setHiddenState - This method will set the hidden state depending on the tab that is clicked.
+     * If the tab is clicked, the tab will be visible, other tabs will be hidden.
+     * @param tab {String} - The tab which should be visible.
+     * @returns {Promise<void>}
+     */
+    async #setHiddenState(tab) {
+        const target = this.#target || document.querySelector(this.getAttribute("for"));
+        const tabs = target.querySelectorAll(":scope > *"); // Get all direct children of the target element.
```

Comment:
now that I think of it. We can use target.children. No need to query selector twice